src/postamble.js

```diff
@@ -353,17 +353,18 @@ if ({{{ ENVIRONMENT_IS_MAIN_THREAD() }}}) {
 // In modularize mode the generated code is within a factory function so we
 // can use await here (since it's not top-level-await).
 wasmExports = await createWasm();
+run();
 #else
 // With async instantation wasmExports is assigned asynchronously when the
 // instance is received.
-createWasm();
+createWasm().then(() => run());
```

Comment:
IIUC this will delay preRun (including preloading data files) until the wasm is ready, but I think before we could do them in parallel?

Reply:
Maybe, but we already that just above where we `await createWasm`.  I will check..